XiEditor/XiCore.swift

```diff
@@ -0,0 +1,46 @@
+// Copyright 2016 The xi-editor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+protocol XiCore {
+    func clientStarted(configDir: String?, clientExtrasDir: String?)
+    func setTheme(themeName: String)
+    func tracingConfig(enabled: Bool)
```

Comment:
I might also include the `newView` method now, because it's synchronous and so has a few other requirements; implementing it might highlight some things you would need to change to support that?

Reply:
It seems synchronous, but in the end it's implemented using [dispatchWithCallback](https://github.com/xi-editor/xi-mac/blob/master/XiEditor/XiDocumentController.swift#L126).
I added a simple implementation with tests using `RpcCallback`.

Reply:
Okay, cool, can we also replace current use of `Events.NewView` with the new methods? If we need to just go through the appdelegate directly for now that's fine.

Reply:
Done! I tried to avoid some force unwrapping along the way. Also..
Here it is: [link](https://github.com/xi-editor/xi-mac/pull/326/commits/0a84df12fc35374c0c7c5438a69bfa79d123cae7)

Reply:
Done! I tried to avoid some force unwrapping along the way. Is it fine for you?